cli/binary/switch.go

```diff
@@ -0,0 +1,149 @@
+package binary
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/fatih/color"
+	"github.com/manifoldco/promptui"
+	"github.com/tarantool/tt/cli/config"
+	"github.com/tarantool/tt/cli/search"
+	"github.com/tarantool/tt/cli/util"
+	"github.com/tarantool/tt/cli/version"
+)
+
+// SwitchCtx contains information for switch command.
+type SwitchCtx struct {
+	// ProgramName is a program name to switch to.
+	ProgramName string
+	// Version of the program to switch to.
+	Version string
+}
+
+// ansi is a string to clean strings from ANSI escape codes using regexp.
+const ansi = "[\u001B\u009B][[\\]()#;?]*(?:(?:(?:[a-zA-Z\\d]*(?:;" +
+	"[a-zA-Z\\d]*)*)?\u0007)|(?:(?:\\d{1,4}(?:;\\d{0,4})*)?[\\dA-PRZcf-ntqry=><~]))"
+
+func cleanString(str string) string {
+	var re = regexp.MustCompile(ansi)
+	return re.ReplaceAllString(str, "")
+}
+
+// ChooseProgram shows a menu in terminal to choose program for switch.
+func ChooseProgram(supportedPrograms []string) (string, error) {
+	programSelect := promptui.Select{
+		Label:        "Select program",
+		Items:        supportedPrograms,
+		HideSelected: true,
+	}
+	_, program, err := programSelect.Run()
+
+	return program, err
+}
+
+// ChooseVersion shows a menu in terminal to choose version of program to switch to.
+func ChooseVersion(cliOpts *config.CliOpts, programName string) (string, error) {
```

Comment:
nit: `ChooseVersion` uses only `cliOpts.Env.BinDir` from `cliOpts`. Does it make sense to pass it as argument to remove `binary` dependency from `config`?